### PR TITLE
Add JSON tests covering joins with different join key permutations.

### PR DESCRIPTION
**Pull request by big-andy-coates**
### Description 

Add JSON test cases covering joins where:
 - the join key is not present in the projection
 - both left and right join keys are present in the projection
 - only the right join key is present in the projection

Note: I've had to split some existing tests into explicit AVRO and JSON versions because the schema for internal topics differs between the two. (Annoying!).

Note: I've had to change some existing tests to use old style names, i.e. set `ksql.named.internal.topics` to `off`, (and hence `ksql.streams.topology.optimization` to `none`) as I wanted to enhance the test to also check the messages in the internal topics. To ensure consistent topic names across all versions the test must use the old style names.  Changing this required regenerating the expect topology files.

I've also added expected topology files for the new tests to the 5_3_pre dir.

### Testing done 
Test only change.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksql-engine/src/test/resources/query-validation-tests/joins.json [hunk 1]
```diff
@@ -148,6 +148,105 @@
...
+          {"name": "LEFT_OUTER_JOIN", "type": "stream", "keyField": {"name": null, "legacyName": "T.ID", "legacySchema": "BIGINT"}}
+        ]
+      }
+    },
+    {
+      "name": "stream stream left join - both join keys in projection",
+      "statements": [
+        "CREATE STREAM TEST (ID bigint, NAME varchar, VALUE bigint) WITH (kafka_topic='left_topic', value_format='JSON');",
+        "CREATE STREAM TEST_STREAM (ID bigint, F1 varchar, F2 bigint) WITH (kafka_topic='right_topic', value_format='JSON');",
+        "CREATE STREAM LEFT_OUTER_JOIN as SELECT t.id, tt.id, name, value, f1, f2 FROM test t left join TEST_STREAM tt WITHIN 11 seconds ON t.id = tt.id;"
```
Comment (agavra) [inline ksql-engine/src/test/resources/query-validation-tests/joins.json]: what is the logic to select one vs the other? what if we re-arranged the select clauses?

Comment (big-andy-coates) [reply ksql-engine/src/test/resources/query-validation-tests/joins.json]: The order shouldn't matter. It should always prefer the left key.

File: ksql-engine/src/test/resources/query-validation-tests/joins.json [hunk 2]
```diff
@@ -397,112 +631,390 @@
...
+        {"topic": "INNER_JOIN", "key": 0, "value": {"NAME": "bar", "VALUE": 99, "F1": "a", "F2": 10}, "timestamp": 16000}
+      ],
+      "post": {
+        "sources": [
+          {"name": "INNER_JOIN", "type": "table", "keyField": {"name": null, "legacyName": null}}
+        ]
+      }
+    },
+    {
+      "name": "table table inner join - right join key in projection",
```
Comment (vcrfxia) [inline ksql-engine/src/test/resources/query-validation-tests/joins.json]: There don't seem to be tests where both keys are in the project for the table table inner join, table table outer join, stream table left join and stream table inner join cases. Not sure how comprehensive you were aiming to be, but thought I'd throw it out there.

Comment (big-andy-coates) [reply ksql-engine/src/test/resources/query-validation-tests/joins.json]: Yeah, I didn't add those as the logic for choosing the key is in one place...

**Pull request merged**